### PR TITLE
[WIP] Viewing Past Orders

### PR DESCRIPTION
**Pull request by stevepentler**
Implementing projects, project developer join table

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -4,11 +4,15 @@ class ApplicationController < ActionController::Base
...
-  helper_method :current_user?
+  helper_method :current_user?, :current_contractor
 
   def current_pending_tribe
+    @ptribe = current_pending_tribe_prep
+  end
+
+  def current_pending_tribe_prep
     session[:tribe] ||= []
-    @ptribe = PendingTribe.new(session[:tribe])
```
Comment (afg419) [inline app/controllers/application_controller.rb]: Nice.  Like pulling this out to a class.

File: app/controllers/projects_controller.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+class ProjectsController < ApplicationController
+  def create
+    @developers = @ptribe.developers
+    total = @ptribe.total 
+    session[:tribe] = []
+    @project = current_contractor.projects.create(project_params.merge({total: total}))
+    @project.developers += @developers 
+    redirect_to contractor_project_path(@project)
```
Comment (afg419) [inline app/controllers/projects_controller.rb]: We're going to need to clean this up.  Duh.  Have you ever heard of pushing logic down the stack?

File: config/routes.rb [hunk 3]
```diff
@@ -4,10 +4,15 @@
   resources :developers, only: [:index, :show]
   resource :tribe, only: [:create, :show, :destroy]
   resources :specialties, only:[:show]
-  resource :contractor, only: [:create, :show]
+  resource :contractor, only: [:create, :show] do 
+    resources :projects, only: [:create, :show] do
+      get 'payment', to: "projects#submit_payment"
```
Comment (afg419) [inline config/routes.rb]: Sweet.

File: app/controllers/projects_controller.rb (hunk 2 above)
Comment (stevepentler) [inline app/controllers/projects_controller.rb]: No.

**Comment by afg419** [comment]
We're going to need tests for changing status to "paid" when an order is purchased.

**Pull request merged**